bccsp/pkcs11/pkcs11.go

```diff
@@ -800,6 +817,19 @@ func (csp *Provider) ecPoint(session pkcs11.SessionHandle, key pkcs11.ObjectHand
 	return ecpt, oid, nil
 }
 
+func (csp *Provider) handleSessionReturn(err error, session pkcs11.SessionHandle) {
+	if err != nil {
+		for _, sessionErr := range sessionErrors {
+			if strings.Contains(err.Error(), sessionErr) {
```

Comment:
why not extract the error code from the string, e.g: "0xB1:" and then do a map lookup?

Reply:
Also, it seems that the only errors you care about, and all the errors you care about, [start with a B](https://github.com/miekg/pkcs11/blob/287d9350987cc9334667882061e202e96cdfb4d0/error.go#L54-L61), so why not just assert that it is the case?